### PR TITLE
Activity Feed

### PR DESCRIPTION
**Pull request by StevenAb**
Created servlet and jsp for the activity feed and updated web.xml to handle "/activityfeed" urls.

Sort of underestimated the amount of code I wrote during this session and ended up dumping the entire prototype into this PR; sorry! Will submit smaller code segments for review in the future.

Also- should we be linking our prototype pages in the navigation bars of existing jsps?

### CODE REVIEW

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Sets the UserStore used by this servlet.
+     * This function provides a common setup method for use by the test framework or servlet's init() function.
+     */
+    void setUserStore(UserStore userStore) {
+        this.userStore = userStore;
+    }
+    
+    /**
+     * This function firex when a user navigates to the activity feed page.
```
Comment (erschwartz) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: firex -> fires

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     * It forwards the request to activityfeed.jsp
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+        throws IOException, ServletException {
+            request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
```
Comment (erschwartz) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: I'd rename to activity-feed.jsp

File: src/main/webapp/WEB-INF/view/activityfeed.jsp [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
```
Comment (erschwartz) [inline src/main/webapp/WEB-INF/view/activityfeed.jsp]: See comment about renaming this file

File: src/main/webapp/WEB-INF/view/activityfeed.jsp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+<html>
+<head>
+    <title>Activity Feed</title>
+    <link rel="stylesheet" href="/css/main.css">
+</head>
+<body>
+
+    <nav>
```
Comment (erschwartz) [inline src/main/webapp/WEB-INF/view/activityfeed.jsp]: Write a to-do here please

format:
`TODO($USERNAME): Take out the trash`

**Review by erschwartz — CHANGES_REQUESTED** [review]
No worries, this is perfect size.

To answer your question: I would only link the prototypes once they are in a more finished state (i.e. there are no major to-dos left).

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setUserStore(UserStore.getInstance());
+    }
+    
+    /**
+     * Sets the UserStore used by this servlet.
+     * This function provides a common setup method for use by the test framework or servlet's init() function.
```
Comment (MatthewOakley) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: For line 29. I don't know if this is just me but could you break this up into two lines so its easier to read.

**Review by MatthewOakley — APPROVED** [review]
This looks good to me. That little change isn't needed it just makes it easier to read.

**Comment by StevenAb** [comment]
Hey, just fixed the comments, changed the filename, and added a todo!

**Review by MatthewOakley — APPROVED** [review]
This looks good to me!

**Comment by erschwartz** [comment]
LGTM

**Review by erschwartz — APPROVED** [review]

**Pull request merged**